### ISSUE TITLE
Incorrect `WPS70x` logic

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#985]
# Bug report

## What's wrong

It now works with strings. And give false positives a lot.

Example:

```python
some = call('`Optional[None]` found and Literal[None]')
```

Gives:

```
  1:1      WPS700 Found useless `Literal[None]` typing annotation
  some = call('`Optional[None]` found and Literal[None]')
  ^

  1:1      WPS703 Found `Optional[None]` which is just `None`
  some = call('`Optional[None]` found and Literal[None]')
```

We should rethink our methods.

Related #976
